.changeset/wise-apples-call.md

```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/react-testing': major
+---
+
+WHAT: Release act promises when root wrapper is destroyed
+WHY: To prevent unresolved promises from failing subsequent act calls
+HOW: `destroyAll` in `afterEach` will now have to be awaited
```

Comment:
This should also call out that `destroy` is now async (not just `destroyAll`) since it's also a public API of Root